File: 2_Python/Algorithm/Sorting/selectionsort.py
def selection_sort(array):
    """
    Selection sort implementation

    :param array: unsorted array
    :return: None
    """

    size = len(array)

    for i in range(size-1):

        index = i

        for j in range(i+1, size):

            if array[index] > array[j]:

                index = j

        array[i], array[index] = array[index], array[i]

File: 5_Go/algorithm/sort/selectionsort.go
package sort

func SelectionSort(arr []int) {
	size := len(arr)
	for i := 0; i < size-1; i++ {
		index := i
		for j := i+1; j < size; j++ {
			if arr[index] > arr[j] {
				arr[j], arr[index] = arr[index], arr[j]
			}
		}
		smallerNumber := arr[index]
		arr[index] = arr[i]
		arr[i] = smallerNumber
	}
}
